docs/network-layout-spec.md

```diff
@@ -0,0 +1,172 @@
+
+      # Where to deploy nodes for this group. Note that nodes are numbered
+      # according to the order in which the regions appear here.
+      regions:
+        - us_east_1: 2    # my_validators[0], my_validators[1]
+        - us_east_2: 1    # my_validators[2]
+        - us_west_1: 3    # my_validators[3], my_validators[4], my_validators[5]
+
+  - my_seeds:
+      tendermint: v0.31.7
+      config_template: ./seednode-config.toml
+      # We want these nodes to be seeds
+      seeds: yes
+      # Where to deploy seed nodes for this group
+      regions:
+        - us_east_1: 1    # my_seeds[0]
+        - us_west_1: 1    # my_seeds[1]
+
+  - late_joiners:
+      # If you want to deploy a custom Tendermint binary, specify its path
+      # instead of an official release version.
+      tendermint: /path/to/local/tendermint
+
+      # We don't want these nodes to be included from the beginning. You will
+      # have to add them to the network later manually/programmatically.
+      in_genesis: no
+
+      # The nodes should not be started automatically (assumes that you will
+      # start the Tendermint services manually at a later stage)
+      started: no
+
+      config_template: ./late-joiners-config.toml
+      use_seeds:
+        - my_seeds
+      regions:
+        - us_east_1: 1   # late_joiners[0]
+
+load_tests:
+  - load0:
+      # In future, `tm-load-test` will be supported. This will influence all of
+      # the following parameters.
+      method: tm-bench
+
+      # The number of VMs (running tm-bench) to start
+      client_nodes: 1
+
+      # A list of targets for this load test
+      targets:
+        # Can specify an entire group to connect to all endpoints
+        - my_validators
+        # Can specify a single node within a group
+        - late_joiners[0]
+      
+      # The number of seconds for which to run the load test
+      time: 120
+      broadcast_tx_method: async
+      connections: 1
+      # The number of transactions per second to generate
+      rate: 1000
+      # The number of bytes to generate per transaction
+      size: 250
+```
+
+NOTES:
+1. The configuration file here needs to allow for easy environment variable
+   interpolation. The BASH convention is probably best, e.g. `"$MYVAR"`.
+
+## Interaction
+In order to be easy to use, the test network layout tool would have to
+facilitate:
+
+1. Deployment of the network
+2. Destruction of the network
+3. Starting/stopping of individual Tendermint nodes
+4. Start/stop load tests
+
+For example, one should simply be able to run something like:
+
+```bash
+# By default, the tool should look for a file called `tmtestnet.yaml` or
+# `tmtestnet.yml` in the current folder. Override with --config parameter.
+
+# Deploy the network (creating all AWS resources)
+tmtestnet network deploy
+
+# Destroy the entire network (deleting all AWS resources)
+tmtestnet network destroy
+
+# Start specific node(s)
+tmtestnet network start node4 node5 node6
```

Comment:
This would be used to start the late joiners? How do we know which node is "node5"?

Reply:
At first my thinking was that the node IDs would be assigned in the order in which they're defined. On second thought though, this may not be optimal. Let me see if there's possibly a better naming scheme here and will submit some changes.